### PR TITLE
Feature : Page/content listing

### LINKED ISSUE #82: [CMS] Content listing page

### PR DESCRIPTION
**Pull request by simonjulianl**
finish content listing for desktop and mobile, Resolves #82

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -3,6 +3,21 @@ module.exports = {
   darkMode: false, // or 'media' or 'class'
   purge: false,
   theme: {
+    minWidth: {
+      0: "0",
```
Comment (jamesyaputra) [inline tailwind.config.js:7]: Why did you need to declare new width configurations?

File: tailwind.config.js [hunk 2]
```diff
@@ -38,6 +54,23 @@ module.exports = {
       height: () => ({
         "screen-75": "75vh",
       }),
+      width: {
+        "1/7": "14.2857143%",
```
Comment (jamesyaputra) [inline tailwind.config.js:58]: Same goes for this (width configs)

File: tailwind.config.js [hunk 3]
```diff
@@ -12,6 +27,7 @@ module.exports = {
       admissions: "url('/assets/backgrounds/admissions.jpg')",
       events: "url('/assets/backgrounds/events.jpg')",
       contact: "url('/assets/backgrounds/contact.jpg')",
+      aksaraBox: "url('/assets/backgrounds/aksaraBox.png')",
```
Comment (jamesyaputra) [inline tailwind.config.js:30]: Instead of using preloaded assets, can you change the `AksaraBox` component to render images from URLs? Cause each article will have its own image sent from the DB

File: src/pageContent/Aksara/index.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                  title={item.title}
+                  description={item.description}
+                  hyperlink="https://aksarapinus.wordpress.com/"
+                ></AksaraBox>
+              </div>
+            );
+          })}
+        </div>
+        <div className={`mt-2 lg-min:mt-0 lg-min:mb-10`}>
+          <Button variant="secondary" onClick={() => router.push("/about")}>
```
Comment (jamesyaputra) [inline src/pageContent/Aksara/index.tsx:64]: Can make it such that when you click it actually loads more.

Limit the render to just 3 boxes first until user clicks `Load More`

File: src/components/AksaraBox/index.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import Text from "src/components/Text";
+
+interface OwnProps {
+  bgImage?: string;
+  title: string;
+  description: string;
+  hyperlink: string;
+}
+
+const AksaraBox: React.FC<OwnProps> = ({
```
Comment (jamesyaputra) [inline src/components/AksaraBox/index.tsx:12]: Can rename to something like `ContentPreview`

File: src/components/Navbar/index.tsx [hunk 6]
```diff
@@ -15,10 +15,11 @@ export function renderNavLink(
 ): React.ReactFragment {
   const { title, slug } = link;
   const pagePaths = pathname.split("/");
+  const isAksara = pagePaths[1] === "aksara";
```
Comment (jamesyaputra) [inline src/components/Navbar/index.tsx:18]: Instead of doing this, I suggest doing this instead:

In the `Navbar/links` file, can add a new element to the links objects
something like this `{title: "Aksara", slug: "aksara", alt: true}`

If alt is true, render Navbar with alternate colors/themes

This is to make the component as generic and as uncoupled from specificities like slug names as much as possible

File: src/components/Page/index.tsx [hunk 7]
```diff
@@ -52,13 +52,15 @@ const Page: React.FC<OwnProps> = ({
...
       </Head>
       <Navbar pathname={router.pathname} />
-      <Banner
-        title={title}
-        description={description}
-        bgImage={bgImage}
-        subBanner={subBanner}
-        renderSubcontent={renderSubcontent}
-      />
+      {title != "Aksara" ? (
```
Comment (jamesyaputra) [inline src/components/Page/index.tsx:55]: Instead of doing this, can add a new prop to let the component know whether to render `Banner` or not

Again, this is to keep the component as generic as possible

File: src/components/Text/index.tsx [hunk 8]
```diff
@@ -15,6 +15,10 @@ const textStyles = {
   body: { size: "xl", weight: "normal" },
   "body-small": { size: "lg", weight: "normal" },
   subtext: { size: "base", weight: "normal" },
+  aksaraHeader: { size: "4xl", weight: "extrabold" },
```
Comment (jamesyaputra) [inline src/components/Text/index.tsx:18]: Can avoid naming new text styles with `aksara` prefixed
Instead, you can rename the whole text styles to something like body, body-alt, header, header-alt, etc.

This will require some files to be changed as they specify the text style, so will need your help with that!

**Review by jamesyaputra — CHANGES_REQUESTED** [review]

File: src/components/AksaraBox/index.tsx (hunk 5 above)
Comment (simonjulianl) [reply src/components/AksaraBox/index.tsx:12]: done

File: src/components/Navbar/index.tsx (hunk 6 above)
Comment (simonjulianl) [reply src/components/Navbar/index.tsx:18]: done

**Review by simonjulianl — COMMENTED** [review]
clicked lol

File: src/components/Page/index.tsx (hunk 7 above)
Comment (simonjulianl) [reply src/components/Page/index.tsx:55]: done

File: src/components/Page/index.tsx [hunk 9]
```diff
@@ -14,6 +14,10 @@ interface OwnProps {
   renderSubcontent?: () => React.ReactNode;
   children: React.ReactNode;
   router: Router;
+<<<<<<< HEAD
```
Comment (jamesyaputra) [inline src/components/Page/index.tsx:17]: Can you remove these conflicts?

File: src/redux/post.tsx [hunk 10]
```diff
@@ -124,10 +153,10 @@ const postsSlice = createSlice({
     },
     [getPostsById.fulfilled.type]: (state, action) => {
       const newPost = action.payload;
-      const index = state.posts.findIndex((post) => post.id === newPost.id);
-      if (index === -1) {
-        state.posts.push(newPost);
-      }
+      // const index = state.posts.findIndex((post) => post.id === newPost.id);
```
Comment (jamesyaputra) [inline src/redux/post.tsx:156]: Why are these commented out?

File: src/components/AksaraBox/index.tsx (hunk 5 above)
Comment (jamesyaputra) [inline src/components/AksaraBox/index.tsx:12]: Jangan lupa didelete

**Review by jamesyaputra — CHANGES_REQUESTED** [review]

File: src/components/Page/index.tsx [hunk 11]
```diff
@@ -24,6 +25,7 @@ const Page: React.FC<OwnProps> = ({
   renderSubcontent,
   children,
   router,
+  unrenderBanner,
```
Comment (jamesyaputra) [inline src/components/Page/index.tsx:28]: can do `renderBanner = true` instead, then for aksara just set false

**Review by jamesyaputra — CHANGES_REQUESTED** [review]

**Review by jamesyaputra — APPROVED** [review]
